src/services/userService.js

```diff
@@ -0,0 +1,76 @@
+const jwt = require("jsonwebtoken");
+const axios = require("axios");
+
+const userDao = require("../models/userDao");
+const { throwCustomError } = require("../utils/error");
+
+const SocialTypeId = Object.freeze({
+  KAKAO: 1,
+  NAVER: 2,
+  GOOGLE: 3,
+});
+
+const kakaoLogin = async (authCode) => {
+  const getKakaoToken = await axios.get("https://kauth.kakao.com/oauth/token", {
+    headers: {
+      "Content-Type": "application/x-www-form-urlencoded",
+    },
+    params: {
+      grant_type: process.env.GRANT_TYPE,
+      client_id: process.env.REST_API_KEY,
+      redirect_url: process.env.REDIRECT_URI,
+      code: authCode,
+    },
+    withCredentials: true,
+  });
+
+  if (!getKakaoToken) throwCustomError("MISSING_KAKAO_TOKEN", 400);
+
+  const getKakaoUserData = await axios.get(
+    "https://kapi.kakao.com/v2/user/me",
+    {
+      headers: {
+        Authorization: `Bearer ${getKakaoToken.data.access_token}`,
+      },
+    }
+  );
+
+  if (!getKakaoUserData.data.id)
+    throwCustomError("CANNOT_GET_KAKAO_USER_DATA", 400);
+
+  const socialId = getKakaoUserData.data.id;
+  const email = getKakaoUserData.data.kakao_account.email;
+  const nickname = getKakaoUserData.data.properties.nickname;
+  const socialTypeId = SocialTypeId.KAKAO;
+
+  const [userInfo] = await userDao.getUserInfo(socialId, socialTypeId);
```

Comment:
```suggestion
  const userInfo = await userDao.getUserInfo(socialId, socialTypeId);
```
이전에 dao layer에서 드린 코멘트가 적용된다면 위와같이 수정해주세요~